hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -135,10 +136,12 @@
 
 	private DefaultSizeStrategy defaultSizeStrategy;
 
+	private final Integer version;
```

Comment:
wrapper type is chosen for some dialect doesn't support 'version' and we store `null` value to denote that.